src/storage/txn/actions/check_txn_status.rs

```diff
@@ -153,10 +160,13 @@ pub fn rollback_lock(
     lock: &Lock,
     is_pessimistic_txn: bool,
     collapse_rollback: bool,
+    for_flashback: bool,
 ) -> Result<Option<ReleasedLock>> {
     let overlapped_write = match reader.get_txn_commit_record(&key)? {
         TxnCommitRecord::None { overlapped_write } => overlapped_write,
-        TxnCommitRecord::SingleRecord { write, .. } if write.write_type != WriteType::Rollback => {
+        TxnCommitRecord::SingleRecord { write, .. }
+            if write.write_type != WriteType::Rollback && !for_flashback =>
```

Comment:
Why is it possible for flashback to roll back some lock with a committed record?

Reply:
Here is the case:

- Lock `self.start_key` with flashback's `start_ts`.
- Flashback `self.start_key` to write a new record with flashback's `commit_ts`.
- Rolling back the lock `self.start_key` will see the newly written flashback committed record.

Reply:
It breaks how transaction normally works so it makes me feel a bit worried.

Can you ensure `start_key` is not a real key, so there can't be a new record written to the key?

Reply:
Okay, let me have a try on this way.

Reply:
Or, I think you can make the `start_key` a bit special. Other keys are flashbacked in a 1PC way, except for this key with a lock, which is committed like a normal 2PC. It's committed until the very end of the process.

Reply:
Done. PTAL.